pandas/tests/frame/methods/test_replace.py

```diff
@@ -1363,3 +1363,83 @@ def test_replace_after_convert_dtypes(self):
         result = df.replace(1, 10)
         expected = pd.DataFrame({"grp": [10, 2, 3, 4, 5]}, dtype="Int64")
         tm.assert_frame_equal(result, expected)
+
+    def test_replace_real_dont_change_other_columns(self, any_real_dtype):
+        # GH30512
+        dtype = any_real_dtype
+        value_a = 2
+        value_b = 3
+        df = pd.DataFrame({"A": [value_a], "B": [value_b]}, dtype=dtype)
+        result = df.replace(to_replace={value_a: None}).dtypes.drop("A")
```

Comment:
can you do the .dtypes.drop separately?  the `df.replace` call is the one we care about